### PR TITLE
[12.0][MIG] migrate sale order line description

### PR DESCRIPTION
**Pull request by mohamedhagag**
migrate sale order line description to v 12.0

### CODE REVIEW

**Comment by mohamedhagag** [comment]
@aheficent OK I got the reason for the error in your test:
You are trying to add a line without selecting a partner, I will fix it .

**Comment by mohamedhagag** [comment]
@aheficent I fixed the issue in my code
But I found that the below module does not handle such issue:
sale_milestone_profile_invoicing
I uninstalled the mentioned module and tried the workflow and it's working fine.

**Comment by AaronHForgeFlow** [comment]
Ok, so another fix has to be done to that module sale_milestone_profile_invoicing @TDu 

Thanks for the clarification @mohamedhagag

**Comment by mohamedhagag** [comment]
@TDu @aheficent 

I just made a PR for sale_milestone_profile_invoicing #896 

waiting for the build to test

**Review by AaronHForgeFlow — APPROVED** [review]
Functional tests good to me.

File: sale_order_line_description/models/sale_order_line.py [hunk 1]
```diff
@@ -16,6 +16,8 @@ def product_id_change(self):
         if (self.user_has_groups(
                 'sale_order_line_description.'
                 'group_use_product_description_per_so_line') and
-                self.product_id.description_sale):
-            self.name = self.product_id.description_sale
+                self.product_id.description_sale) and self.order_id.partner_id:
```
Comment (pedrobaeza) [inline sale_order_line_description/models/sale_order_line.py]: You shouldn't put the partner as condition here, but only for filling the lang context retrieving the field contents. If no partner, no context, but field must be retrieved.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
And please squash your migration commits together

File: sale_order_line_description/models/sale_order_line.py (hunk 1 above)
Comment (mohamedhagag) [reply sale_order_line_description/models/sale_order_line.py]: I added the partner to avoid the error mentioned by @aheficent - will remove it .

Comment (pedrobaeza) [reply sale_order_line_description/models/sale_order_line.py]: No, you have to do a middle solution.

Comment (mohamedhagag) [reply sale_order_line_description/models/sale_order_line.py]: sorry but I tried to squash my commits but I failed to do.
How to squash them ? sorry for that .

**Comment by mohamedhagag** [comment]
I made the commit squash, I'm not sure if this is what you mean .

**Comment by pedrobaeza** [comment]
Yes the squash is correct, but the code remains the same. I will suggest proper code.

File: sale_order_line_description/models/sale_order_line.py [hunk 2]
```diff
@@ -17,5 +17,7 @@ def product_id_change(self):
                 'sale_order_line_description.'
                 'group_use_product_description_per_so_line') and
                 self.product_id.description_sale):
-            self.name = self.product_id.description_sale
+            self.name = self.product_id.with_context(
```
Comment (pedrobaeza) [inline sale_order_line_description/models/sale_order_line.py]: Proper code for this:
```python
product = self.product_id
if self.order_id.partner_id:
    product = product.with_context(
        lang=self.order_id.partner_id.lang,
    )
self.name = product.description_sale

Comment (mohamedhagag) [reply sale_order_line_description/models/sale_order_line.py]: done, thanks for guidance

**Review by pedrobaeza — APPROVED** [review]
Although the 2 commits must be squashed

**Pull request closed without merging**